services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/SlotProcessor.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.services.beaconchain;
+
+import static com.google.common.primitives.UnsignedLong.ONE;
+import static com.google.common.primitives.UnsignedLong.ZERO;
+import static tech.pegasys.teku.datastructures.util.BeaconStateUtil.compute_epoch_at_slot;
+import static tech.pegasys.teku.datastructures.util.BeaconStateUtil.compute_start_slot_at_epoch;
+import static tech.pegasys.teku.util.config.Constants.SECONDS_PER_SLOT;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.eventbus.EventBus;
+import com.google.common.primitives.UnsignedLong;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.core.ForkChoiceUtil;
+import tech.pegasys.teku.datastructures.blocks.NodeSlot;
+import tech.pegasys.teku.logging.EventLogger;
+import tech.pegasys.teku.networking.eth2.Eth2Network;
+import tech.pegasys.teku.statetransition.events.attestation.BroadcastAggregatesEvent;
+import tech.pegasys.teku.statetransition.events.attestation.BroadcastAttestationEvent;
+import tech.pegasys.teku.statetransition.forkchoice.ForkChoice;
+import tech.pegasys.teku.storage.client.RecentChainData;
+import tech.pegasys.teku.sync.SyncService;
+import tech.pegasys.teku.util.time.channels.SlotEventsChannel;
+
+public class SlotProcessor {
+  private final RecentChainData recentChainData;
+  private final SyncService syncService;
+  private final ForkChoice forkChoice;
+  private final Eth2Network p2pNetwork;
+  private final SlotEventsChannel slotEventsChannelPublisher;
+  private final EventBus eventBus;
+  private final NodeSlot nodeSlot = new NodeSlot(ZERO);
+  private final EventLogger EVENT_LOG;
```

Comment:
nit: lower case as it's not a constant
```suggestion
  private final EventLogger eventLog;
```